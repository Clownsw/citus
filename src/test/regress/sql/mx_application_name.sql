CREATE SCHEMA mx_app_name;

CREATE TABLE output (line text);

-- a hack to run command with another user
\COPY output FROM PROGRAM 'PGAPPNAME=mx_app_name psql -c "CREATE TABLE dist_1(a int)" -h localhost -p 57636 -U postgres -d regression'
\COPY output FROM PROGRAM 'PGAPPNAME=mx_app_name psql -c "SELECT create_distributed_table(''dist_1'', ''a'');" -h localhost -p 57636 -U postgres -d regression'

-- ensure the command executed fine
SELECT count(*) FROM pg_dist_partition WHERE logicalrelid = 'dist_1'::regclass;

\COPY output FROM PROGRAM 'PGAPPNAME=mx_app_name psql -c "DROP TABLE dist_1;" -h localhost -p 57636 -U postgres -d regression'

DROP SCHEMA mx_app_name CASCADE;
