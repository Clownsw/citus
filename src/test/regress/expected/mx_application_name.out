CREATE SCHEMA mx_app_name;
CREATE TABLE output (line text);
-- a hack to run command with another application name
\COPY output FROM PROGRAM 'psql postgres://postgres@localhost:xxxxx/regression?application_name=test -c "CREATE TABLE dist_1(a int)"'
\COPY output FROM PROGRAM 'psql postgres://postgres@localhost:xxxxx/regression?application_name=test -c "SELECT create_distributed_table(''dist_1'', ''a'');"'
-- ensure the command executed fine
SELECT count(*) FROM pg_dist_partition WHERE logicalrelid = 'dist_1'::regclass;
 count
---------------------------------------------------------------------
     1
(1 row)

\COPY output FROM PROGRAM 'psql postgres://postgres@localhost:xxxxx/regression?application_name=test -c "DROP TABLE dist_1;"'
DROP SCHEMA mx_app_name CASCADE;
